webui/react/src/pages/ExperimentDetails.tsx

```diff
@@ -116,8 +128,93 @@ const ExperimentDetailsComp: React.FC = () => {
     setForkModalState(state => ({ ...state, visible: false }));
   };
 
+  const handleCheckpointShow = (event: React.MouseEvent, checkpoint: CheckpointDetail) => {
+    event.stopPropagation();
+    setActiveCheckpoint(checkpoint);
+    setShowCheckpoint(true);
+  };
+  const handleCheckpointDismiss = () => setShowCheckpoint(false);
+
+  const columns: ColumnsType<TrialItem> = [
```

Comment:
suggestion: define the table in another component.

Reply:
Yeah makes sense to do this. Especially if we move into moving one-off components into a pages subdirectory. Let's leave this core work to a separate ticket.